packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts

```diff
@@ -10,48 +10,72 @@ import {
+import type { EIP1193Provider, Hash, PublicClient, TypedDataDefinition } from "viem";
 import { Hex, createWalletClient, custom, http } from "viem";
+import type { Chain, HttpTransport, Transport } from "viem";
 import { Web3 } from "web3";
 
 import { EVMBlockchainIncludingTestnet } from "@crossmint/common-sdk-base";
 
 import { CrossmintWalletService } from "../../api/CrossmintWalletService";
-import { getBundlerRPC, getPaymasterRPC, getUrlProviderByBlockchain, getViemNetwork } from "../BlockchainNetworks";
+import { TChain, entryPoint, getBundlerRPC, getPaymasterRPC, getUrlProviderByBlockchain, getViemNetwork } from "../BlockchainNetworks";
 import { Custodian } from "../plugins";
 import { TokenType } from "../token";
 
 export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchainIncludingTestnet> {
     private sessionKeySignerAddress?: Hex;
     private crossmintService: CrossmintWalletService;
     private publicClient: PublicClient;
-    private ecdsaValidator: KernelValidator<"ECDSAValidator">;
-    private account: KernelSmartAccount;
-    private kernelClient: KernelAccountClient<Transport, Chain, KernelSmartAccount>;
+    private ecdsaValidator: KernelValidator<entryPoint, "ECDSAValidator">;
+    private account: KernelSmartAccount<EntryPoint, HttpTransport, TChain>;
+    private kernelClient: KernelAccountClient<
+        entryPoint,
+        HttpTransport,
+        TChain,
+        KernelSmartAccount<entryPoint, HttpTransport, TChain>
+    >;
     chain: B;
 
     constructor(
-        account: KernelSmartAccount,
+        account: KernelSmartAccount<EntryPoint, HttpTransport, TChain>,
         crossmintService: CrossmintWalletService,
         chain: B,
         publicClient: PublicClient,
-        ecdsaValidator: KernelValidator<"ECDSAValidator">
+        ecdsaValidator: KernelValidator<entryPoint, "ECDSAValidator">
     ) {
         this.chain = chain;
         this.crossmintService = crossmintService;
         this.publicClient = publicClient;
         this.ecdsaValidator = ecdsaValidator;
         this.kernelClient = createKernelAccountClient({
+            account: account as any,
+            entryPoint: ENTRYPOINT_ADDRESS_V07,
+            chain: getViemNetwork(chain),
+            bundlerTransport: http(getBundlerRPC(chain)),
+            middleware: {
+                sponsorUserOperation: async ({ userOperation }) => {
+                    const paymasterClient = createZeroDevPaymasterClient({
+                        chain: getViemNetwork(chain),
+                        transport: http(getPaymasterRPC(chain)),
+                        entryPoint: ENTRYPOINT_ADDRESS_V07,
+                    });
+                    return paymasterClient.sponsorUserOperation({
+                        userOperation,
+                        entryPoint: ENTRYPOINT_ADDRESS_V07,
+                    });
+                },
+            },
+        }) as any;
+        /*this.kernelClient = createKernelAccountClient({
```

Comment:
Let's remove all commented code. It will create noise in the long run and we already have it in the branch tree